### PR TITLE
Fixed window function math for tapering function

### PR DESCRIPTION
**Pull request by acliu**
This fixes the window functions to account for tapering. Currently it assumes a frequency-independent primary beam, but fixing this will require some changes to pyuvdata that haven't been implemented yet.

### CODE REVIEW

**Comment by nkern** [comment]
the `get_G` call in `PSpecData.pspec` doesn't get the `taper` kwarg propagated..

**Comment by bhazelton** [comment]
Out of curiosity, what do you need from `pyuvdata` that doesn't currently exist?

**Comment by acliu** [comment]
Right now we have something that returns Omega_pp(\nu_i) ~ \int A^2 (\nu_i) dOmega. Turns out we'll need Omega_pp(\nu_i, \nu_j) ~ \int A(\nu_i) A(\nu_j) dOmega as well.

**Comment by bhazelton** [comment]
What does the i & j index? different antennas, frequencies, or something else?

**Comment by acliu** [comment]
Different frequencies.

**Comment by acliu** [comment]
Ok, just pushed. Hopefully this time it's a little less buggy.

I mentioned to @nkern today that in working through this, I realized that the pspec scalar normalization  infrastructure is not currently able to accommodate non-trivial tapers + non-trivial weighting matrices simultaneously. Details in the notes that I've attached, which will be posted to the HERA memo series.

There *is* a hacky way to do this, involving assigning half (i.e., the square root) of the weighting matrix to Q and having another square root of the weight act on the data. But this is beginning to feel like epicycles upon epicycles, so if there is a need to push further along this direction I'm ripping out the old stuff and implementing Q "properly" (i.e., I'll subclass and make a version where Q is truly equal to dC/dp).

[main.pdf](https://github.com/HERA-Team/hera_pspec/files/1969547/main.pdf)

File: hera_pspec/pspecdata.py [hunk 1]
```diff
@@ -681,6 +685,75 @@ def get_G(self, key1, key2):
                 G[i,j] += np.einsum('ab,ba', iR1Q[i], iR2Q[j])
 
         return G / 2.
+
+    def get_H(self, key1, key2, taper='none'):
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: I don't see `H_ab` defined anywhere in the note?

File: hera_pspec/pspecdata.py [hunk 2]
```diff
@@ -681,6 +685,75 @@ def get_G(self, key1, key2):
...
+            <q_a> = \sum_b H_{ab} p_b
+
+        This is given by
+
+            H_ab = (1/2) Tr[R_1 Q_a^alt R_2 Q_b]
+
+        (See HERA memo #44). As currently implemented, this approximates the
+        primary beam as frequency independent. Under this approximation, the
+        our H_ab is defined using the equation above *except* we have
+        Q^{yet_another} rather than Q_b, where
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: `yet_another` is not a great name for a variable...

File: hera_pspec/pspecdata.py [hunk 3]
```diff
@@ -750,54 +826,61 @@ def get_MW(self, G, mode='I'):
...
         # Check that mode is supported
         modes = ['G^-1', 'G^-1/2', 'I', 'L^-1']
         assert(mode in modes)
 
         # Build M matrix according to specified mode
         if mode == 'G^-1':
-            M = np.linalg.pinv(G, rcond=1e-12)
-            #U,S,V = np.linalg.svd(F)
-            #M = np.einsum('ij,j,jk', V.T, 1./S, U.T)
+            raise NotImplementedError
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: Please say what's not implemented, e.g. `NoteImplementedError("G^-1 mode not currently supported.")`

**Comment by philbull** [comment]
(OK, I now see that you do raise warnings about this.)

File: hera_pspec/pspecdata.py (hunk 1 above)
Comment (acliu) [reply hera_pspec/pspecdata.py]: Actually, it's there. Equation 33. Unfortunate page break location makes it hard to see.

File: hera_pspec/pspecdata.py (hunk 2 above)
Comment (acliu) [reply hera_pspec/pspecdata.py]: Changed it to `Q^{tapered}`

File: hera_pspec/pspecdata.py (hunk 3 above)
Comment (acliu) [reply hera_pspec/pspecdata.py]: Done!

**Pull request merged**